src/regexbench.h

```diff
@@ -48,13 +49,23 @@ namespace regexbench {
 class Engine;
 class PcapSource;
 
+struct ResultInfo {
+  ResultInfo() : nmatches(0), nmatched_pkts(0), npkts(0), nbytes(0) {}
+  size_t nmatches;
+  size_t nmatched_pkts;
+  size_t npkts;
+  size_t nbytes;
+};
+
 struct MatchResult {
-  MatchResult() : nmatches(0), nmatched_pkts(0) {}
+  MatchResult() : stop(false) {}
 
   struct timeval udiff;
   struct timeval sdiff;
-  size_t nmatches;
-  size_t nmatched_pkts;
+  struct ResultInfo cur;
+  struct ResultInfo old;
+  bool stop;
```

Comment:
How about declaring this as `std::atomic_bool` and use corresponding API's where it's being used.